### PR TITLE
fix: online event and enhance date/time in front page

### LINKED ISSUE #5926: Front Page: Add info about Online event and fix/enhance date/time
The front page nicely displays locations of events. At the same place there should be
* the info "Online Event" if the event is not location based and 
* "Online Event and at [address here]" if it is a mixed event

Furthermore the tile needs to be enhanced in regards to the date to
* show long date type for the event 
* show both start and end time of event (as many events go over several days now)

Screenshot from 2020-12-06 01-03-57

### PR DESCRIPTION
**Pull request by MeghalBisht**
Fixes #5926 

#### Short description of what this resolves:
1. the info "Online Event" if the event is not location based and
2. "Online Event and at [address here]" if it is a mixed event
3. show long date type for the event
4. show both start and end time of event 

**Screenshots -**

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/templates/components/event-card.hbs [hunk 1]
```diff
@@ -23,10 +23,18 @@
...
+          {{general-date this.event.startsAt 'date-time-long'}}
+          -
+          {{general-date this.event.endsAt 'date-time-long'}}
         </span>
       </div>
       <SmartOverflow @class="description">
+        {{#if this.event.online}}
+          Online Event
+        {{/if}}
+        {{#if (and this.event.online this.event.shortLocationName)}}
```
Comment (daretobedifferent18) [inline app/templates/components/event-card.hbs:35]: use this if statement inside upper if statement and then no need to add ```and``` just use like 
```handlebars
{{#if this.event.shortLocationName}}
       ...
       ...
{{/if}}
```

Comment (MeghalBisht) [reply app/templates/components/event-card.hbs:35]: if the event is not online then location will also not be displayed.

Comment (MeghalBisht) [reply app/templates/components/event-card.hbs:35]: thanks 😄

**Review by daretobedifferent18 — APPROVED** [review]
For me it's working fine

File: app/templates/components/event-card.hbs [hunk 2]
```diff
@@ -23,10 +23,18 @@
       </SmartOverflow>
       <div class="meta">
         <span class="date">
-          {{general-date this.event.startsAt tz=this.event.timezone}}
```
Comment (iamareebjamal) [inline app/templates/components/event-card.hbs:26]: Why do you think we had `tz=this.event.timezone`. We had not written it for fun, there was a reason for it. Why did you remove it?

File: app/templates/components/event-card.hbs [hunk 3]
```diff
@@ -23,10 +23,18 @@
...
+          {{general-date this.event.startsAt 'date-time-long'}}
+          -
+          {{general-date this.event.endsAt 'date-time-long'}}
         </span>
       </div>
       <SmartOverflow @class="description">
+        {{#if this.event.online}}
+          Online Event
```
Comment (iamareebjamal) [inline app/templates/components/event-card.hbs:33]: Wrap in t

File: app/templates/components/event-card.hbs [hunk 4]
```diff
@@ -23,10 +23,18 @@
...
+          {{#if this.event.shortLocationName}}
+            and at
```
Comment (iamareebjamal) [inline app/templates/components/event-card.hbs:35]: wrap in t

File: app/templates/components/event-card.hbs [hunk 5]
```diff
@@ -23,10 +23,18 @@
       </SmartOverflow>
       <div class="meta">
         <span class="date">
-          {{general-date this.event.startsAt tz=this.event.timezone}}
+          {{general-date this.event.startsAt 'date-time-long'}}
```
Comment (daretobedifferent18) [inline app/templates/components/event-card.hbs:26]: also change date-time-long -> date-time-tz-long

Comment (daretobedifferent18) [reply app/templates/components/event-card.hbs:26]: @iamareebjamal shouldn't we do this also?

Comment (iamareebjamal) [reply app/templates/components/event-card.hbs:26]: yes

**Comment by iamareebjamal** [comment]
@mariobehling I think it'll look better this way and separate information as well

image
image
image

**Comment by mariobehling** [comment]
@iamareebjamal Nice! So, if you want us to choose a version, I would choose version 2. Better do not introduce another green color.

@MeghalBisht Could you please change the dates in a way: When the beginning and end of the event is on the same day just keep the date first and e.g. 2:00 PM - 5:00 PM (CET).

Thank you!

**Comment by MeghalBisht** [comment]
**After update -**

File: app/templates/components/event-card.hbs [hunk 6]
```diff
@@ -17,18 +17,37 @@
...
     {{/unless}}
-    <a class="main content" href="{{href-to (if this.isCFS 'public.cfs' 'public') this.event.identifier}}">
+    <a style="min-height:165px" class="main content" href="{{href-to (if this.isCFS 'public.cfs' 'public') this.event.identifier}}">
       <SmartOverflow @class="header">
         {{this.event.name}}
       </SmartOverflow>
       <div class="meta">
         <span class="date">
-          {{general-date this.event.startsAt tz=this.event.timezone}}
+          {{#if (eq (general-date this.event.startsAt 'date-short') (general-date this.event.endsAt 'date-short')) }}
```
Comment (daretobedifferent18) [inline app/templates/components/event-card.hbs:26]: No, don't do it like this

Comment (daretobedifferent18) [reply app/templates/components/event-card.hbs:26]: Instead check for whether it ```isSingleDay``` in event-card.js file and then use it here

File: app/templates/components/event-card.hbs [hunk 7]
```diff
@@ -17,18 +17,37 @@
...
+            {{general-date this.event.startsAt 'date-time-long'}}
+            - 
+            {{general-date this.event.endsAt 'time-tz-short' tz=this.event.timezone}}
+          {{else}}
+            {{general-date this.event.startsAt 'date-time-tz-long' tz=this.event.timezone}}
```
Comment (daretobedifferent18) [inline app/templates/components/event-card.hbs:31]: Also here show timezone only with endAt date

File: app/templates/components/event-card.hbs [hunk 8]
```diff
@@ -17,18 +17,37 @@
...
-      <SmartOverflow @class="description">
-        {{this.event.shortLocationName}}
-      </SmartOverflow>
+      {{#if this.event.online}}
+        <p style="color: #000000AD">
+          <i class="wifi icon"></i>
+          {{t 'Online Event'}}
+        </p>
+      {{/if}}
+      <p style="color: #000000AD">
```
Comment (daretobedifferent18) [inline app/templates/components/event-card.hbs:43]: put this entire thing inside 
```handlebars
{{#if}}
     ....
{{/if}}
```

File: app/templates/components/event-card.hbs (hunk 6 above)
Comment (MeghalBisht) [reply app/templates/components/event-card.hbs:26]: okay, just wanted to know is this wrong?

File: app/templates/components/event-card.hbs [hunk 9]
```diff
@@ -23,12 +23,12 @@
...
       <div class="meta">
         <span class="date">
-          {{#if (eq (general-date this.event.startsAt 'date-short') (general-date this.event.endsAt 'date-short')) }}
+          {{#if this.isSingleDay}}
             {{general-date this.event.startsAt 'date-time-long'}}
             - 
             {{general-date this.event.endsAt 'time-tz-short' tz=this.event.timezone}}
           {{else}}
-            {{general-date this.event.startsAt 'date-time-tz-long' tz=this.event.timezone}}
+            {{general-date this.event.startsAt 'date-time-long'}}
```
Comment (daretobedifferent18) [inline app/templates/components/event-card.hbs:31]: why removed tz=this.event.timezone

File: app/templates/components/event-card.hbs [hunk 10]
```diff
@@ -23,12 +23,12 @@
       </SmartOverflow>
       <div class="meta">
         <span class="date">
-          {{#if (eq (general-date this.event.startsAt 'date-short') (general-date this.event.endsAt 'date-short')) }}
+          {{#if this.isSingleDay}}
             {{general-date this.event.startsAt 'date-time-long'}}
```
Comment (daretobedifferent18) [inline app/templates/components/event-card.hbs:27]: pls understand passing tz is not to show tz, it is used in general-date to get time in particular tz

**Review by daretobedifferent18 — COMMENTED** [review]
Screenshot from 2020-12-08 02-47-30
still showing tz on both dates

File: app/templates/components/event-card.hbs (hunk 10 above)
Comment (MeghalBisht) [reply app/templates/components/event-card.hbs:27]: Thanks for clarifying.

**Comment by iamareebjamal** [comment]
Screenshot_20201208-035834~2
The card is elongated even though there's a lot of place below the address. Please ensure that there is no extra space below the address in case of online+address events so that the card occupies least height as possible

**Comment by MeghalBisht** [comment]
> The card is elongated even though there's a lot of place below the address. Please ensure that there is no extra space below the address in case of online+address events so that the card occupies least height as possible

Please clarify, all cards need to be of the same height?

**Comment by iamareebjamal** [comment]
Of course!

**Comment by MeghalBisht** [comment]
> Of course!
So for that max-height should be hard coded(for mobile & desktop)? As currently height is variable depending on the contents inside the card

**Comment by iamareebjamal** [comment]
Currently they are all of same height, this should remain same after changes. To achieve this, remove the empty space below address

**Comment by MeghalBisht** [comment]
> Currently they are all of same height, this should remain same after changes. To achieve this, remove the empty space below address

This is not equal. Pls check

**Comment by iamareebjamal** [comment]
After reducing the size, it is not equal, that is another issue but at least on max size it is equal and thus should remain after this PR
image

**Comment by iamareebjamal** [comment]
image

Add margin top of 0.5em on online event and this event is not a single day event and yet it shows up as a single day event

**Comment by MeghalBisht** [comment]
> Add margin top of 0.5em on online event and this event is not a single day event and yet it shows up as a single day event

 isSingleDay() {
    return this.startsAt.isSame(this.endsAt, 'day');
  }
This is used to check for same day, I cannot understand why the above case is happening. Am i missing something?

File: app/components/event-card.js [hunk 11]
```diff
@@ -39,4 +39,8 @@ export default class EventCard extends Component {
   selectEventType(eventType) {
     this.set('eventType', eventType === this.eventType ? null : eventType);
   }
+
+  isSingleDay() {
+    return this.startsAt.isSame(this.endsAt, 'day');
```
Comment (daretobedifferent18) [inline app/components/event-card.js:44]: here it should be ```this.event.startsAt``` and ```this.event.endsAt```

Comment (MeghalBisht) [reply app/components/event-card.js:44]: same result

Comment (iamareebjamal) [reply app/components/event-card.js:44]: Maybe it is checking `day` as in Friday instead of date

File: app/templates/components/event-card.hbs [hunk 12]
```diff
@@ -17,11 +17,11 @@
           @src={{if this.event.thumbnailImageUrl this.event.thumbnailImageUrl this.event.originalImageUrl}} />
       </a>
     {{/unless}}
-    <a style="min-height:165px" class="main content" href="{{href-to (if this.isCFS 'public.cfs' 'public') this.event.identifier}}">
+    <a class="{{if this.device.isMobile '' 'card-height'}} main content" href="{{href-to (if this.isCFS 'public.cfs' 'public') this.event.identifier}}">
       <SmartOverflow @class="header">
         {{this.event.name}}
       </SmartOverflow>
-      <div class="meta">
+      <div class="meta" style="margin-bottom: 0.5em">
```
Comment (iamareebjamal) [inline app/templates/components/event-card.hbs:24]: Use mb- class

**Review by maze-runnar — APPROVED** [review]

File: app/styles/libs/_helpers.scss [hunk 13]
```diff
@@ -98,3 +98,7 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
 .no-wrap {
   white-space: nowrap;
 }
+
+.card-height {
+  min-height: 165px;
```
Comment (iamareebjamal) [inline app/styles/libs/_helpers.scss:103]: Is this needed?

Comment (MeghalBisht) [reply app/styles/libs/_helpers.scss:103]: Yes, else in dekstop view the cards will not be of equal size ( online + location events have greater height). I have removed this in mobile view so extra height is not added there

**Comment by MeghalBisht** [comment]
@iamareebjamal  I have used isSame function in this PR from momentjs, just saw this warning message in their docs, should this be changed?
https://momentjscom.readthedocs.io/en/latest/moment/05-query/02-is-same/

**Comment by iamareebjamal** [comment]
No
